File: moq-python/subscriber.py
import asyncio
import moq

async def main():
    async with moq.Client("https://cdn.moq.dev/anon") as client:
        async for announcement in client.announced("my-stream"):
            # Subscribe to the raw "chat" track
            track = await announcement.broadcast.subscribe_track("chat")
            print("Subscribed. Waiting for messages...")

            # Iterate groups, then frames within each group
            async for group in track:
                async for frame in group:
                    message = frame.payload.decode("utf-8")
                    print(f"Received: {message}")

asyncio.run(main())

File: moq-ts/subscriber.ts
// subscriber.js
import * as Moq from "@moq/net";

async function main() {
    // 1. Connect to the same relay
    const connection = await Moq.Connection.connect(
        new URL("https://cdn.moq.dev/anon")
    );

    // 2. Consume the broadcast and subscribe to the "chat" track
    const consumer = connection
        .consume(Moq.Path.from("my-stream"))
        .track("chat")
        .subscribe({ priority: 0 });

    console.log("Subscribed. Waiting for messages...");

    // 3. Read incoming groups in a loop
    for (;;) {
        const group = await consumer.recvGroup();
        if (!group) break; // Stream ended

        const message = await group.readString();
        console.log(`Received: ${message}`);
    }
}

main().catch(console.error);    